internal/storage/v2/grpc/tracereader_test.go

```diff
@@ -90,12 +100,108 @@ func startServer(t *testing.T, server *grpc.Server, listener net.Listener) *grpc
 	return conn
 }
 
+func makeTestTrace() ptrace.Traces {
+	trace := ptrace.NewTraces()
+	resources := trace.ResourceSpans().AppendEmpty()
+	scopes := resources.ScopeSpans().AppendEmpty()
+
+	spanA := scopes.Spans().AppendEmpty()
+	spanA.SetName("foobar")
+	spanA.SetTraceID(pcommon.TraceID([16]byte{1}))
+	spanA.SetSpanID(pcommon.SpanID([8]byte{2}))
+	spanA.SetKind(ptrace.SpanKindServer)
+	spanA.Status().SetCode(ptrace.StatusCodeError)
+
+	return trace
+}
+
 func TestTraceReader_GetTraces(t *testing.T) {
-	tr := &TraceReader{}
+	tests := []struct {
+		name           string
+		testServer     *testServer
+		traces         []*jptrace.TracesData
+		expectedTraces []ptrace.Traces
+		expectedError  string
+	}{
+		{
+			name: "single trace",
+			testServer: &testServer{
+				traces: func() []*jptrace.TracesData {
+					trace := makeTestTrace()
+					traces := []*jptrace.TracesData{(*jptrace.TracesData)(&trace)}
+					return traces
+				}(),
+			},
+			expectedTraces: []ptrace.Traces{makeTestTrace()},
+		},
+		{
+			name: "multiple traces",
+			testServer: &testServer{
+				traces: func() []*jptrace.TracesData {
+					traceA := makeTestTrace()
+					traceB := makeTestTrace()
+					traces := []*jptrace.TracesData{
+						(*jptrace.TracesData)(&traceA),
+						(*jptrace.TracesData)(&traceB),
+					}
+					return traces
+				}(),
+			},
+			expectedTraces: []ptrace.Traces{makeTestTrace(), makeTestTrace()},
+		},
+		{
+			name: "error",
+			testServer: &testServer{
+				traces: func() []*jptrace.TracesData {
+					trace := ptrace.NewTraces()
+					traces := []*jptrace.TracesData{(*jptrace.TracesData)(&trace)}
+					return traces
+				}(),
+				err: assert.AnError,
+			},
+			expectedError: "received error from grpc stream",
+		},
+	}
 
-	require.Panics(t, func() {
-		tr.GetTraces(context.Background(), tracestore.GetTraceParams{})
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			conn := startTestServer(t, test.testServer)
```

Comment:
if you do this here (depending on input flag), then you will reproduce `received error from grpc reader client` error:
```
conn2, err := grpc.NewClient(":0",
  grpc.WithTransportCredentials(insecure.NewCredentials()),
)
require.NoError(t, err)
reader := NewTraceReader(conn2)
```

Reply:
@yurishkuro Awesome! Thanks so much.